gateway-js/src/index.ts

```diff
@@ -375,40 +317,89 @@ export class ApolloGateway implements GraphQLService {
       };
     }
 
-    // Before @apollo/gateway v0.23, ApolloGateway didn't expect stop() to be
-    // called after it started. The only thing that stop() did at that point was
-    // cancel the poll timer, and so to prevent that timer from keeping an
-    // otherwise-finished Node process alive, ApolloGateway unconditionally
-    // called unref() on that timeout. As part of making the ApolloGateway
-    // lifecycle more predictable and concrete (and to allow for a future where
-    // there are other reasons to make sure to explicitly stop your gateway),
-    // v0.23 tries to avoid calling unref().
-    //
-    // Apollo Server v2.20 and newer calls gateway.stop() from its stop()
-    // method, so as long as you're using v2.20, ApolloGateway won't keep
-    // running after you stop your server, and your Node process can shut down.
-    // To make this change a bit less backwards-incompatible, we detect if it
-    // looks like you're using an older version of Apollo Server; if so, we
-    // still call unref(). Specifically: Apollo Server has always passed an
-    // options object to load(), and before v2.18 it did not pass the `apollo`
-    // key on it. So if we detect that particular pattern, we assume we're with
-    // pre-v2.18 Apollo Server and we still call unref(). So this will be a
-    // behavior change only for:
-    // - non-Apollo-Server uses of ApolloGateway (where you can add your own
-    //   call to gateway.stop())
-    // - Apollo Server v2.18 and v2.19 (where you can either do the small
-    //   compatible upgrade or add your own call to gateway.stop())
-    // - if you don't call stop() on your ApolloServer (but in that case other
-    //   things like usage reporting will also stop shutdown, so you should fix
-    //   that)
-    const unrefTimer = !!options && !options.apollo;
-
     this.maybeWarnOnConflictingConfig();
 
     // Handles initial assignment of `this.schema`, `this.queryPlanner`
-    isStaticConfig(this.config)
-      ? this.loadStatic(this.config)
-      : await this.loadDynamic(unrefTimer);
+    if (isStaticSupergraphSdlConfig(this.config)) {
+      const supergraphSdl = this.config.supergraphSdl;
+      await this.initializeSupergraphManager({
+        initialize: async () => {
+          return {
+            supergraphSdl,
+          };
+        },
+      });
```

Comment:
Most of these have dedicated `SupergraphManager`s, this one is simple and just written inline.